processing/src/main/java/org/apache/druid/data/input/impl/AggregateProjectionSpec.java

```diff
@@ -218,10 +223,9 @@ private static ProjectionOrdering computeOrdering(VirtualColumns virtualColumns,
 
     String timeColumnName = null;
     Granularity granularity = null;
-    // try to find the __time column equivalent, which might be a time_floor expression to model granularity
-    // bucketing. The time column is decided as the finest granularity on __time detected. If the projection does
-    // not have a time-like column, the granularity will be handled as ALL for the projection and all projection
-    // rows will use a synthetic timestamp of the minimum timestamp of the incremental index
+
+    // determine the granularity and time column name for the projection, based on the first time-like grouping column.
+    // if there are multiple time-like grouping columns, they must be "coarser" than the first time-like grouping column.
```

Comment:
this is too restrictive, the __time column does not need to be first, we just need to replace any candidate __time column with the finer granularity column, since what we are looking for is the finest granularity column we can use as a substitute for __time in any queries against the base table